### PR TITLE
Add kha structures

### PR DESCRIPTION
**Pull request by kevinresol**

### CODE REVIEW

File: hxmath/math/MathTypes.hx [hunk 1]
```diff
@@ -157,6 +157,188 @@ abstract QuaternionType(h3d.Quat) from h3d.Quat to h3d.Quat
...
+    private inline function set_m23(v:Float):Float return this._23 = v;
+
+    public var m33(get, set):Float;
+    private inline function get_m33():Float return this._33;
+    private inline function set_m33(v:Float):Float return this._33 = v;
+}
+
+abstract QuaternionType(kha.math.Quaternion) from kha.math.Quaternion to kha.math.Quaternion
+{
+
```
Comment (tbrosman) [inline hxmath/math/MathTypes.hx:328]: Minor: extra newline

Comment (kevinresol) [reply hxmath/math/MathTypes.hx:328]: well this is copied from the h3d type, so at least they are consistent

File: hxmath/math/MathTypes.hx [hunk 2]
```diff
@@ -157,6 +157,188 @@ abstract QuaternionType(h3d.Quat) from h3d.Quat to h3d.Quat
...
+    public inline function new(s:Float, x:Float, y:Float, z:Float)
```
Comment (tbrosman) [inline hxmath/math/MathTypes.hx:329]: It is probably worth documenting that Kha's Quaternion class has the real portion last (whereas QuaternionType has it as the first parameter).

**Comment by tbrosman** [comment]
Looks good! I left a few comments, but saw no functional issues.

File: hxmath/math/MathTypes.hx (hunk 1 above)
Comment (tbrosman) [reply hxmath/math/MathTypes.hx:328]: Ah, fair enough. I'll make a pass later and clean them up at the same time.

**Pull request merged**